testing/solomachine.go

```diff
@@ -231,119 +346,74 @@ func (solo *Solomachine) ChanCloseConfirm(chain *TestChain, channelID string) {
 	require.NotNil(solo.t, res)
 }
 
-// CreateHeader generates a new private/public key pair and creates the
-// necessary signature to construct a valid solo machine header.
-// A new diversifier will be used as well
-func (solo *Solomachine) CreateHeader(newDiversifier string) *solomachine.Header {
-	// generate new private keys and signature for header
-	newPrivKeys, newPubKeys, newPubKey := GenerateKeys(solo.t, uint64(len(solo.PrivateKeys)))
-
-	publicKey, err := codectypes.NewAnyWithValue(newPubKey)
-	require.NoError(solo.t, err)
-
-	data := &solomachine.HeaderData{
-		NewPubKey:      publicKey,
-		NewDiversifier: newDiversifier,
-	}
-
-	dataBz, err := solo.cdc.Marshal(data)
-	require.NoError(solo.t, err)
-
-	signBytes := &solomachine.SignBytes{
-		Sequence:    solo.Sequence,
-		Timestamp:   solo.Time,
-		Diversifier: solo.Diversifier,
-		Path:        []byte{},
-		Data:        dataBz,
-	}
-
-	bz, err := solo.cdc.Marshal(signBytes)
-	require.NoError(solo.t, err)
-
-	sig := solo.GenerateSignature(bz)
-
-	header := &solomachine.Header{
-		Sequence:       solo.Sequence,
-		Timestamp:      solo.Time,
-		Signature:      sig,
-		NewPublicKey:   publicKey,
-		NewDiversifier: newDiversifier,
-	}
-
-	// assumes successful header update
-	solo.Sequence++
-	solo.PrivateKeys = newPrivKeys
-	solo.PublicKeys = newPubKeys
-	solo.PublicKey = newPubKey
-	solo.Diversifier = newDiversifier
-
-	return header
+// SendPacket mocks sending a packet by setting a packet commitment directly.
+func (solo *Solomachine) SendPacket(chain *TestChain, packet channeltypes.Packet) {
+	commitmentHash := channeltypes.CommitPacket(chain.Codec, packet)
+	chain.GetSimApp().IBCKeeper.ChannelKeeper.SetPacketCommitment(chain.GetContext(), packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence(), commitmentHash)
```

Comment:
Ideally we actually send like an ics20 transfer. There was a bug in ibc core channel keeper in relation to solution machine which wouldn't be caught by this, but would be caught by calling ics20 send. I think other clients might look at these tests as inspiration, so might be useful to have a full test. Totally fine with creating a followup issue

Reply:
Great suggestion, I'll create a follow up issue for this 👍